travis_ci.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+
+set -e # fast fail
+
+# action for travis ci
+Action=$1
+# config the number of integration test job.
+JOBS=$2
+# config the id of job
+JOB_ID=$3
+
+CASE_PREFIX="testcase.list"
+LOCAL_CASE_FILE="${TRAVIS_BUILD_DIR}/test/${CASE_PREFIX}"
+
+source "${TRAVIS_BUILD_DIR}/hack/install/config.sh"
+
+prepare_testcase() {
+  # get the number of all test case
+  # test case must write with this format:
+  # func (XXX) TestXXX(c *check.C) {
+  grep -h -E "^func\ \(.*\)\ Test.*\(c\ \*check\.C\)\ \{" ./test -r \
+    | awk '{print $3,$4}' | awk -F \( '{print $1}' \
+    | sed 's/\*//' | sed 's/) /./' > "${LOCAL_CASE_FILE}"
+
+  # make test case list file
+  local sum nums
+  local index=1
+  local loop=1
+
+  # shellcheck disable=SC2002
+  sum=$(cat "${LOCAL_CASE_FILE}" | wc -l)
+  nums=$((sum / JOBS + 1))
+
+  rm -rf "${LOCAL_CASE_FILE}.*"
+  # shellcheck disable=SC2013
+  for test in $(cat "${LOCAL_CASE_FILE}"); do
+    tmp=$((loop * nums))
+    if [[ ${index} -gt ${tmp} ]]; then
+      loop=$((loop + 1))
+    fi
+    index=$((index + 1))
+    echo "${test}" >> "${LOCAL_CASE_FILE}.${loop}"
+  done
+}
+
+run_pre_test() {
+  prepare_testcase
+}
+
+run_unittest() {
+  sudo env "PATH=$PATH" hack/install/install_ci_related.sh
+  make unit-test
+  make coverage
+
+  bash <(curl -s https://codecov.io/bash) -cF unittest -y .codecov.yml
+}
+
+run_integration_test() {
+  local job_id=$1
+
+  sudo systemctl stop docker
```

Comment:
if there is not docker.service?

Reply:
will be failed.... so I need check it before...